enterprise/frontend/src/metabase-enterprise/tenants/EditUserStrategyModal/EditUserStrategyModal.tsx

```diff
@@ -90,6 +85,14 @@ export const EditUserStrategyModal = ({
 
     setIsApplyingAfterConfirm(false);
     onClose();
+
+    if (selectedStrategy === "multi-tenant") {
+      // Wait for settings to be refetched before navigating.
+      // This ensures `createTenantsRouteGuard` sees the updated setting.
+      await refetch();
```

Comment:
Without this, the route guard thinks that tenants is not yet enabled and pushes us back to the people page. The `await` lets us ensure that `use-tenants` cache are refreshed.